libs/pika/command_line_handling/src/command_line_handling.cpp

```diff
@@ -398,7 +399,9 @@ namespace pika::detail {
     std::size_t handle_mpi_completion_mode(
         detail::manage_config& cfgmap, pika::program_options::variables_map& vm)
     {
-        std::size_t completion_mode = cfgmap.get_value<std::size_t>("pika.mpi.completion_mode", 0);
+        int def_val = pika::detail::get_env_var_as<std::size_t>("PIKA_MPI_COMPLETION_MODE", 0);
+        std::size_t completion_mode =
+            cfgmap.get_value<std::size_t>("pika.mpi.completion_mode", def_val);
```

Comment:
I think this should not be needed, but did you see a change in behaviour after you added this? Note the default value for `pika.mpi.completion_mode` is already set in https://github.com/pika-org/pika/blob/50c60c37ef513bce9091826853e9c7fa4b8b71b1/libs/pika/runtime_configuration/src/runtime_configuration.cpp#L342. The `0` as a default value is a bit misleading. It should really just be an error if there's no default, because there should always be a default in the configuration.

Reply:
The stuff you put in broke the existing logic because you overwrite the completion mode with a new default value, so if I set the completion mode using an ENV var, it was thrown out by the default value set by `cfgmap.get_value<std::size_t>("pika.mpi.completion_mode", def_val);` so I simply made sure that the default value set there was taken from the env var. If nothing is passed, the default value of zero is used (which can be changed to something else). But in the main mpi init, we set the default to zero, hence the use here.

Reply:
See https://github.com/biddisco/pika/pull/2.